File: lib/src/models/SingleComponentResponseOfDestinyCharacterRenderComponent.cs
using System.Runtime.Serialization;

namespace GhostSharp.BungieNetApi.Model
{
	[DataContract]
	public class SingleComponentResponseOfDestinyCharacterRenderComponent{

		/// <summary>
		/// Only really useful if you're attempting to render the character's current appearance in 3D, this returns a bare minimum of information, pre-aggregated, that you'll need to perform that rendering. Note that you need to combine this with other 3D assets and data from our servers.
		/// </summary>
		/// <summary>
		/// Examine the Javascript returned by https://bungie.net/sharedbundle/spasm to see how we use this data, but be warned: the rabbit hole goes pretty deep.
		/// </summary>
		[DataMember(Name="data", EmitDefaultValue=false)]
		public DestinyCharacterRenderComponent Data { get; set; }

		[DataMember(Name="privacy", EmitDefaultValue=false)]
		public ComponentPrivacySetting Privacy { get; set; }

		/// <summary>
		/// If true, this component is disabled.
		/// </summary>
		[DataMember(Name="disabled", EmitDefaultValue=false)]
		public bool Disabled { get; set; }


		public override bool Equals(object input)
        {
            return this.Equals(input as SingleComponentResponseOfDestinyCharacterRenderComponent);
        }

		public bool Equals(SingleComponentResponseOfDestinyCharacterRenderComponent input)
		{
			if (input == null) return false;

			return
				(
                    Data == input.Data ||
                    (Data != null && Data.Equals(input.Data))
                ) &&
				(
                    Privacy == input.Privacy ||
                    (Privacy != null && Privacy.Equals(input.Privacy))
                ) &&
				(
                    Disabled == input.Disabled ||
                    (Disabled != null && Disabled.Equals(input.Disabled))
                ) ;
		}
	}
}

File: pkg-generated/models/SingleComponentResponseOfDestinyCharacterRenderComponent.go
package bungieapigo

type SingleComponentResponseOfDestinyCharacterRenderComponent struct {

    // Only really useful if you're attempting to render the character's current appearance in 3D,
    // this returns a bare minimum of information, pre-aggregated, that you'll need to perform that
    // rendering. Note that you need to combine this with other 3D assets and data from our servers.
    // Examine the Javascript returned by https://bungie.net/sharedbundle/spasm to see how we use
    // this data, but be warned: the rabbit hole goes pretty deep.
    Data DestinyCharacterRenderComponent `json:"data"`

    Privacy ComponentPrivacySetting `json:"privacy"`

    // If true, this component is disabled.
    Disabled bool `json:"disabled"`

}

